core/adapters/eth_tx.go

```diff
@@ -58,16 +58,36 @@ func (e *EthTx) TaskType() models.TaskType {
 // is not currently pending. Then it confirms the transaction was confirmed on
 // the blockchain.
 func (e *EthTx) Perform(input models.RunInput, store *strpkg.Store) models.RunOutput {
+	jr, err := store.FindJobRun(input.JobRunID())
```

Comment:
Hmm, the way I did this before was to add a field to `models.RunInput`. Why not add the jobSpecID there instead of doing another DB lookup in this method? It might be more efficient.

Reply:
I also want the run request info not just the jobSpecID. I could add the three fields I want to RunInput or just the whole jobRun but thats a big change (looks like all adapters would change etc.) and I'm not sure its worth it for the v1 pipeline